piksi_multi_cpp/include/piksi_multi_cpp/device/device.h

```diff
@@ -7,16 +7,17 @@
 #include <string>
 #include <vector>
 
+namespace piksi_multi_cpp {
+
 typedef std::string Identifier;
 typedef std::set<Identifier> Identifiers;
 inline bool identifierEqual(const Identifier& a, const Identifier& b) {
   return a.compare(b) == 0;
 }
 
-namespace piksi_multi_cpp {
 class Device {
  public:
-  enum DeviceType { kUSB = 0 };
+  typedef std::shared_ptr<Device> DevicePtr;
```

Comment:
These I often just name Ptr, as they're never used without the class name (giving Device::Ptr) but up to taste :-)